src/front-end/graphics.js

```diff
@@ -487,187 +530,199 @@ var Render = (function(){
+          p.attr({
             fill: Utils.hex(color[0], color[1], color[2]),
-             "fill-opacity": color.a
-         });
-         parallelogram.drag(move, start, stop)
-     }
-
-
-
-    // helper method that draws bbox around an object
-    function _renderBoundingBox(s, obj) {
-        // render the bbox
-        var sq = s.path(obj.getBBox().path);
-        sq.attr({
             "fill-opacity": 0,
-            stroke: "#000",
-            strokeWidth: 1
-        })
-        return sq
-    }
+            stroke: Utils.hex(color[0], color[1], color[2]),
+            strokeWidth: 2
+          });
+          return p;
+        }
+      }
+
+      /**
+      * Renders a Regular Angle Mark on the canvas
+      * @param       {Snap} s  Snap.svg global object
+      * @param       {JSON} properties JSON object from Haskell server
+      */
+      function _renderRegularArc(s,properties){
+          var radius = properties.r
+          var color = properties.color
+          var x = properties.x
+          var y = properties.y
+          var style = properties.style
+          var angle = properties.angle > 360.0 ? 360.0
+                      - properties.angle : properties.angle //Handle angle > 360
+              angle = angle < 0 ? 360 + angle : angle //Handle angle < 0
+          var rotation = properties.rotation + 90 - angle
+
+          var startAngle = rotation
+          var endAngle = angle < 0 ? (rotation - angle) : (rotation + angle)
+
+          var arcPath = describeArc(x, y, radius, startAngle,endAngle,false);
+          var arc = s.path(arcPath);
+          arc.attr({
+              "fill-opacity": 0,
+              stroke: Utils.hex(color[0], color[1], color[2]),
+              strokeWidth: 2
+          });
+
+          if (style == "wedge") {
+              var pf = describeFullArc(x, y, radius, startAngle, endAngle);
+              var f = s.path(pf);
+              f.attr({
+                  // TODO: hardcoded color value here
+                  fill: Utils.hex(0, 0, 205),
+                  "fill-opacity": 0.6,
+              });
+              var g = s.g(arc,f)
+          } else{
+              return arc;
+          }
+      }
+
+          /**
+          * Renders an angle mark on the canvas
+          * @param       {Snap} s  Snap.svg global object
+          * @param       {JSON} properties JSON object from Haskell server
+          */
+          function _renderArc(s, properties) {
+            if(properties.isRight == "true") {
+              return _renderRightArc(s,properties)
+            } else {
+              return _renderRegularArc(s,properties)
+            }
+          }
+
+          /**
+          * Renders a parallelogram on the canvas
+          * @param       {Snap} s  Snap.svg global object
+          * @param       {JSON} obj JSON object from Haskell server
+          */
+          function _renderParallelogram(s, properties) {
+            [x, y] = [properties.x, properties.y]
```

Comment:
document what (x,y) refers to. a corner (and if so, which one)? the center? some other point?